verified_boot_app.md

```diff
@@ -11,8 +38,109 @@ Src: https://github.com/Kicksecure/kicksecure-wiki-backup/blob/6900fdfbdea8693fd
+
+<!--
+* Vendor neutral: yes, any operating system that wishes to be compatible with the "EFI_ALT" standard/convention is welcome to.
+* Kicksecure specific: no
+* Kicksecure branding: no
+-->
+
+1.4. The application MUST be OS/HW vendor agnostic and NOT subject to any
+     branding.
+
+### Keys and boot options management
+
+<!--
+* Boot compatibility with mainstream operating systems such as Microsoft Windows: Yes, the user-controlled verified boot feature should not restrict the user's freedom to boot any operating system, if they so confirm in their firmware. (This is useful to avoid "Linux-only motherboard" and to keep resale value higher.)
+-->
+
+2.1. The application MUST NOT prevent a user from booting any UEFI-compliant
+     operating system. It MAY prompt the user to grant trust to an unknown key
+     or to disable UEFI Secure Boot for that boot, but it must always offer at
+     least one successful path.
+
+<!--
+* Support for binaries signed by multiple keys: Yes. (UEFI already supports that.) (This is important to allow the distribution or customer to cycle/replace the key in later upgrades or for planned signing key changes.)
+-->
+
+2.2. The application MUST support images singed by multiple keys.
+
+<!--
+* Public key TOFU (trust on first use): yes, if confirmed in the firmware.
+-->
+
+2.3. The application MUST ask the user for confirmation to trust given key
+     on the first use. The application MAY maintain the base of trusted keys
+     separately from UEFI Secure Boot or use existing UEFI Secure Boot
+     databases.
+
+<!--
+* Show key fingerprint: yes - "key fingerprint [key fingerprint]" [info symbol] "What is this? Brief explanation that the user can check with the vendor of this operating system if they wish to trust that key."
+-->
+
+2.4. The application SHOULD present the information about keys and
+     certificates in human-readable form using Issuer's and Subject's Common
+     Name (CN) and Organization (O) or other fields if present. Additionally,
+     the key fingerprint MUST always be displayed.
+
+<!--
+* Microsoft Secure Boot default signing key:
+** included in the firmware by default: yes, required for the purpose of allowing the user to boot Microsoft Windows
+** trusted in the firmware by default: no, express permission by the user is needed. (Because this key could lead to Secure Booting a Linux malware distribution. This would be unwanted by users who wish to use full verified boot with their chosen key.)
+* What this is:
+** A mechanism for users to much more easily choose which keys they trust.
+-->
+
+2.5. The firmware SHOULD include the Microsoft UEFI Secure Boot keys and
+     certificates in the firmware vendor's default set of UEFI Secure Boot
+     keys.
+
+2.6. The firmware vendor's default UEFI Secure Boot key and certificate set
+     SHALL NOT be trusted by default. Trust shall only be given to these keys
+     if explicitly confirmed by the user.
```

Comment:
We need only the second sentence with strong MUST, not just SHALL.

If we want standard use for Linux or Windows, shouldn't the user just tell the application or remove the application?

Reply:
> shouldn't the user just tell the application or remove the application?

Tell what?

Why remove the application?

Reply:
Tell "clear any policy application set and use default one defined by platform firmware".

Or

> Why remove the application?

I want to get back to the trust default keys in the system.

This requirement should be:

```
Trust MUST only be given to these keys if explicitly confirmed by the user.
```

And that is enough in my opinion. It also means that the user could trust the firmware vendors' key if the user wants, and it also covers the first sentence because if the user didn't trust the firmware vendors' keys, they would not be trusted. The first sentence is redundant.

Reply:
Removed obsolete sentence. 8f6adaf